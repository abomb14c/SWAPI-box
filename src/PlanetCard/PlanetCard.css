*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    text-align: center;
    margin:0 auto;
    width: 60%;
}

.planet-card {
background-color: #ffffff;
border: 2px solid #D6D6D6;
display: inline-block;
height: 500px;
left: 15px;
margin: 3px;
position: relative;
vertical-align: top;
width: 340px;
}

.planet-name {
    text-align: center;
    color: rgba(255,255,255);
    margin: none;
    text-shadow: 4px 4px #000;
}

.planet-name-section {
    height: 125px;
    background-image: url('../assets/planets-section-header.jpg');
    background-size: cover;
    padding-top: 50px;
    font-size: 36px;
}

.content-div{
    margin: 15px;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}

.residents{
    font-size: 18px;
}