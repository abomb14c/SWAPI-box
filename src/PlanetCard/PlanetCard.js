import React from 'react';
import './PlanetCard.css';
import PropTypes from 'prop-types';

const PlanetCard = (props) => {
  const{name, terrain, population, climate, residents, findFavoritePlanet, id, category} = props;

  return (
    <article className="planet-card">
      <section className="planet-name-section">
        <h1 className="planet-name">{name}</h1>
        <button onClick={()=> findFavoritePlanet(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Terrain</p>
        <h3 className="description-content">{terrain}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Population</p>
        <h3 className="description-content">{population}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Climate</p>
        <h3 className="description-content">{climate}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Residents</p>
        <h3 className="description-content residents">{residents}</h3>
      </div>
    </article>
  );
     
};

PlanetCard.propTypes = {
  name: PropTypes.string,
  terrain: PropTypes.string,
  population: PropTypes.number,
  climate: PropTypes.string,
  residents: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ]),
  findFavoritePlanet: PropTypes.func,
  id: PropTypes.string,
  category: PropTypes.string
};
 
export default PlanetCard;
