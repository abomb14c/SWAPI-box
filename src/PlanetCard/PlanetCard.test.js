import React from 'react';
import ReactDOM from 'react-dom';
import PlanetCard from './PlanetCard';
import {shallow, mount} from 'enzyme';

it('should match the snapshot', () => {
  let name='Earth';
  let terrain='hills';
  let population= 123;
  let climate= 'shitty';
  let findFavorite= jest.fn();
  let id = "Planet0";
  let residents= "people, animals, bears";
  let category="planets";

  let wrapper= shallow(<PlanetCard name={name}
    terrain={terrain}
    population={population}
    climate={climate}
    findFavoritePlanet={findFavorite}
    id={id}
    category={category}
    residents={residents}
  />);
  expect(wrapper).toMatchSnapshot();
});