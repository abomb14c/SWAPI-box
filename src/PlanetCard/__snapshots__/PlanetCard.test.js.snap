// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    category="planets"
    climate="shitty"
    findFavoritePlanet={[Function]}
    id="Planet0"
    name="Earth"
    population={123}
    residents="people, animals, bears"
    terrain="hills"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="planet-name-section"
>
          <button
                    onClick={[Function]}
          >
                    Favorite
          </button>
          <h1
                    className="planet-name"
          >
                    Earth
          </h1>
</section>,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Terrain
          </p>
          <h3
                    className="description-content"
          >
                    hills
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Population
          </p>
          <h3
                    className="description-content"
          >
                    123
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Climate
          </p>
          <h3
                    className="description-content"
          >
                    shitty
          </h3>
</div>,
        <hr />,
        <div
          className="content-div"
>
          <p
                    className="description-title"
          >
                    Residents
          </p>
          <h3
                    className="description-content residents"
          >
                    people, animals, bears
          </h3>
</div>,
      ],
      "className": "planet-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              Favorite
</button>,
            <h1
              className="planet-name"
>
              Earth
</h1>,
          ],
          "className": "planet-name-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Earth",
              "className": "planet-name",
            },
            "ref": null,
            "rendered": "Earth",
            "type": "h1",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Terrain
</p>,
            <h3
              className="description-content"
>
              hills
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Terrain",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Terrain",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hills",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "hills",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Population
</p>,
            <h3
              className="description-content"
>
              123
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Population",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Population",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 123,
              "className": "description-content",
            },
            "ref": null,
            "rendered": 123,
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Climate
</p>,
            <h3
              className="description-content"
>
              shitty
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Climate",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Climate",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "shitty",
              "className": "description-content",
            },
            "ref": null,
            "rendered": "shitty",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description-title"
>
              Residents
</p>,
            <h3
              className="description-content residents"
>
              people, animals, bears
</h3>,
          ],
          "className": "content-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Residents",
              "className": "description-title",
            },
            "ref": null,
            "rendered": "Residents",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "people, animals, bears",
              "className": "description-content residents",
            },
            "ref": null,
            "rendered": "people, animals, bears",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="planet-name-section"
>
            <button
                        onClick={[Function]}
            >
                        Favorite
            </button>
            <h1
                        className="planet-name"
            >
                        Earth
            </h1>
</section>,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Terrain
            </p>
            <h3
                        className="description-content"
            >
                        hills
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Population
            </p>
            <h3
                        className="description-content"
            >
                        123
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Climate
            </p>
            <h3
                        className="description-content"
            >
                        shitty
            </h3>
</div>,
          <hr />,
          <div
            className="content-div"
>
            <p
                        className="description-title"
            >
                        Residents
            </p>
            <h3
                        className="description-content residents"
            >
                        people, animals, bears
            </h3>
</div>,
        ],
        "className": "planet-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                Favorite
</button>,
              <h1
                className="planet-name"
>
                Earth
</h1>,
            ],
            "className": "planet-name-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Earth",
                "className": "planet-name",
              },
              "ref": null,
              "rendered": "Earth",
              "type": "h1",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Terrain
</p>,
              <h3
                className="description-content"
>
                hills
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Terrain",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Terrain",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hills",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "hills",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Population
</p>,
              <h3
                className="description-content"
>
                123
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Population",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 123,
                "className": "description-content",
              },
              "ref": null,
              "rendered": 123,
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Climate
</p>,
              <h3
                className="description-content"
>
                shitty
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Climate",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Climate",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "shitty",
                "className": "description-content",
              },
              "ref": null,
              "rendered": "shitty",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description-title"
>
                Residents
</p>,
              <h3
                className="description-content residents"
>
                people, animals, bears
</h3>,
            ],
            "className": "content-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Residents",
                "className": "description-title",
              },
              "ref": null,
              "rendered": "Residents",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "people, animals, bears",
                "className": "description-content residents",
              },
              "ref": null,
              "rendered": "people, animals, bears",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
